src/dmd/backend/elem.d

```diff
@@ -3328,63 +3581,68 @@ void elem_print_const(elem *e)
         case TYvptr:
         case TYhptr:
         L1:
-            dbg_printf("%dL ",e->EV.Vlong);
+            printf("%dL ",e.EV.Vlong);
             break;
 
         case TYllong:
         L2:
-            dbg_printf("%lldLL ",(unsigned long long)e->EV.Vllong);
+            printf("%lldLL ",cast(ulong)e.EV.Vllong);
             break;
 
         case TYullong:
-            dbg_printf("%lluLL ",(unsigned long long)e->EV.Vullong);
+            printf("%lluLL ",cast(ulong)e.EV.Vullong);
             break;
 
         case TYcent:
         case TYucent:
-            dbg_printf("%lluLL+%lluLL ", (unsigned long long)e->EV.Vcent.msw, (unsigned long long)e->EV.Vcent.lsw);
+            printf("%lluLL+%lluLL ", cast(ulong)e.EV.Vcent.msw, cast(ulong)e.EV.Vcent.lsw);
             break;
 
         case TYfloat:
-            dbg_printf("%gf ",(double)e->EV.Vfloat);
+            printf("%gf ",cast(double)e.EV.Vfloat);
             break;
         case TYdouble:
         case TYdouble_alias:
-            dbg_printf("%g ",(double)e->EV.Vdouble);
+            printf("%g ",cast(double)e.EV.Vdouble);
             break;
         case TYldouble:
         {
-#if _MSC_VER
-            char buffer[3 + 3 * sizeof(targ_ldouble) + 1];
-            ld_sprint(buffer, 'g', e->EV.Vldouble);
-            dbg_printf("%s ", buffer);
-#else
-            dbg_printf("%Lg ", e->EV.Vldouble);
-#endif
+version (CRuntime_Microsoft)
+{
+            char[3 + 3 * (targ_ldouble).sizeof + 1] buffer = void;
+            static if (is(typeof(e.EV.Vldouble) == real))
+                ld_sprint(buffer.ptr, 'g', longdouble_soft(e.EV.Vldouble));
+            else
+                ld_sprint(buffer.ptr, 'g', longdouble_soft(cast(real)e.EV.Vldouble));
```

Comment:
Wouldn't the cast to `real` be irrelevant for the first case (i.e. if it's a real)?